[Ng Zhen Teng] iP

ngzhenteng opened the pull request.

Commit:
Add Gradle support

ngzhenteng committed:
Level 1 Greet, Echo, Exit

Completed Level 1

ngzhenteng committed:
Level 2. Add, List

Complete

ngzhenteng committed:
Level 3. Mark as Done

Add Task Class to encapsulate tasks' states and behaviours

ngzhenteng committed:
Level 4. ToDos, Events, Deadlines

Add Classes Event, Todo and Deadline which extends Task.

ngzhenteng committed:
A-TextUiTesting

Add commands into input.txt to be run by runtest.bat
Add expected output into EXPECTED.TXT

ngzhenteng committed:
Level 5. Handle Errors

Able to handle error of todo without description and invalid inputs such as 1 done instead of done 1.

ngzhenteng committed:
Level 6. Delete

Able to delete tasks by inputting delete [task number]

ngzhenteng committed:
Level-7. Save

Duke writes to user device

ngzhenteng committed:
Level 8. Dates and Times

Date and Time represented as java LocalDate and LocaTime classes

ngzhenteng committed:
A-MoreOOP

Adhere closer to OOP practices

ngzhenteng committed:
Revert "A-MoreOOP"

This reverts commit a1e256af7e6b1c2320193f97e6a0aa763e228f62.

ngzhenteng committed:
Revert "Revert "A-MoreOOP""

This reverts commit a272fbe46ca817231ad58f4c9e8ff47bf3b1cd59.

ngzhenteng committed:
A-MoreOOP

Adhere closer to OOP Practices

ngzhenteng committed:
A-Packages

Organise classes under duke package

ngzhenteng committed:
A-JUnit

Add JUnit tests DeadlineTest and TaskListTest

ngzhenteng committed:
A-JavaDoc

Add JavaDoc comments

ngzhenteng committed:
A-CodingStandard

Comply with Java coding standard

ngzhenteng committed:
A-JavaDoc

Comments with Java coding standard

ngzhenteng committed:
Level 9. Find

Add find keyword feature

ngzhenteng committed:
Merge branch 'branch-A-CodingStandard'

ngzhenteng committed:
Merge branch 'branch-Level-9'

ngzhenteng committed:
Merge branch 'add-gradle-support'

ngzhenteng committed:
A-Gradle

Add gradle to project

solkim-83 commented on src/main/java/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,25 @@
+package main.java.duke;
+
+public class Parser {
```
Hey I like the structure/ design of the code for Parser!

solkim-83 commented on src/main/java/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,13 @@
+package main.java.duke;
+
+public class DukeException extends Exception{
```
Perhaps, would it be better to create separate class for each error ("Oops" statement) that inherits from this DukeException class? Such that it displays the error message instead in the handleUserInput() method. - Just a suggestion! But it still looks good 👍

ngzhenteng replied:
Sounds great, thank you for the suggestion!

solkim-83 commented on src/main/java/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,122 @@
+            } else {
+                // pass user input to parser
+                if (commandComponents.length == 1) {
+                    invalidInput("  \u2639 OOPS!!! The description of a task cannot be empty.");
```
I meant this kind of error statement in the first comment!

solkim-83 reviewed (commented):
Looks good and I was able to understand your code/ design easily! 
Left a few suggestions for the exception handling for UI class. - Just a suggestion! :)

ngzhenteng committed:
Level 10. GUI

Add Graphical User Interface option for Duke application

kaitlynng commented on src/main/java/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
```
Perhaps you can put a space after Task to comply with coding standards?

kaitlynng replied:
Perhaps put a space after Task to comply with coding standards?

kaitlynng commented on src/main/java/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    public LocalTime stringToLocalTime(String hour, String minutes) {
+        LocalTime lc = LocalTime.parse(hour +":" + minutes);
```
Perhaps put a space between the '+' to comply with coding standards?

ngzhenteng replied:
Yes indeed, thank you kaitlyn!

kaitlynng commented on src/main/java/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+
+    /**
+     * Returns a formatted description of the Deadline task, with the date and time dateline wrapped in parenthesis
+     * @return a formatted description of the Deadline task, with the date and time dateline wrapped in parenthesis
```
Perhaps you can change the description for the return statement, as it does not give any additional information not present in your description!

kaitlynng commented on src/main/java/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,138 @@
+     * @return a string to be displayed as the response on CLI or GUI
+     */
+    public String getResultFromParser(String nextInput) {
+        String[] commandComponents = nextInput.split(" ", 2);
```
Perhaps all of this can be handled by the parser as well, as these are all commands?

ngzhenteng replied:
Yes perhaps, the commandComponents is used to recognise the type of command. Thank you!

kaitlynng reviewed (commented):
Your project looks good and complies with coding standards!

ngzhenteng committed:
Add jar file generation with gradle support

mainClassName set to reflect the location of main class.
We can generate jar files using gradle now.

ngzhenteng committed:
Code Quality

Follow Code Quality Guidelines
Refactor code to have variables with narrower scope, less nested if statements, Simpler logic and S.L.A.P

ngzhenteng committed:
A-CodeQuality Update

Improve Code Quality of Storage class

ngzhenteng committed:
A-Assertions

Add assertions to all classes unrelated to JavaFX to allow runtime verification of code assumptions.

ngzhenteng committed:
A-CodeQuality update 2

Deadline and Event classes follow Code Quality guidelines more closely

ngzhenteng committed:
Merge pull request #1 from ngzhenteng/branch-A-Assertions

A-Assertions

ngzhenteng committed:
Merge branch 'branch-A-CodeQuality' into master

ngzhenteng committed:
Merge pull request #3 from ngzhenteng/master

Merge Master into A-CodeQuality

ngzhenteng committed:
Merge pull request #4 from ngzhenteng/branch-A-CodeQuality

Merge pull request #3 from ngzhenteng/master

ngzhenteng committed:
Bug Fix for Storage.java

Remove FileReader and BufferedReader objects

ngzhenteng committed:
C-Priority

Add support for setting priority to new and existing tasks

ngzhenteng committed:
Merge pull request #5 from ngzhenteng/branch-C-Priority

C-Priority

ngzhenteng committed:
A-BetterGui

Add Customisations to GUI.
Add JUnit Tests

ngzhenteng committed:
A-UserGuide

Add error message information for user guide

ngzhenteng committed:
Ui.png

ngzhenteng committed:
Add Title

Add JavaFX Application Title

ngzhenteng committed:
A-UserGuide

ngzhenteng committed:
Merge branch 'master' of https://github.com/ngzhenteng/ip

ngzhenteng committed:
A-UserGuide Update

ngzhenteng committed:
Merge branch 'master' of https://github.com/ngzhenteng/ip

ngzhenteng committed:
Updates for release version

Add ability for GUI to wrap text for longer lists.
Add javadoc comments for all classes

ngzhenteng committed:
Fix issues with formatting of headers

ngzhenteng committed:
Fix priority command
